pkg-manager/core/src/install/extendInstallOptions.ts

```diff
@@ -37,6 +37,7 @@ export interface StrictInstallOptions {
   linkWorkspacePackagesDepth: number
   lockfileOnly: boolean
   fixLockfile: boolean
+  resetResolutionsWithPreferredVersions: boolean
```

Comment:
Maybe just name it dedupe?
```suggestion
  dedupe: boolean
```

Also, you're saying that this works because of the fixLockfile logic but I don't see the `fixLockfile` set to true anywhere

Reply:
> Maybe just name it dedupe?

Can do! Wasn't sure if the extra verbose name here was that helpful. Thanks for the check.

> Also, you're saying that this works because of the fixLockfile logic but I don't see the `fixLockfile` set to true anywhere

I think I could have been more clear in the last comment.

My understanding of `fixLockfile` is that it can regenerate parts of `pnpm-lock.yaml` if there are any dangling resolutions. For example, a package refers to `foo@1.0.0`, but that version doesn't exist in the lockfile anymore.

Since there's no dangling resolutions, `fixLockfile` isn't needed explicitly since pnpm in general rebuilds the wanted lockfile as it resolves resolves the dependency graph. This should be using the same assumptions the existing `forgetResolutionsOfPrevWantedDeps` function does for `importers`. Admittedly, there could be stronger guarantees of this assumption for `packages`.

I think I could extend the test to make sure a `pnpm install` after `pnpm dedupe` doesn't change the lockfile to better capture the assumption `packages` is regenerated.

Reply:
> Admittedly, there could be stronger guarantees of this assumption for packages.

Here are the parts of `resolve-dependencies` that regenerate `packages`. The `toLockfileDependency` function is fine with `prevSnapshot.dependencies` being a partial. This runs regardless of `fixLockfile`.

https://github.com/pnpm/pnpm/blob/b78774ea0f92ccba8f0ee690d88fe39ccfcec7dd/pkg-manager/resolve-dependencies/src/updateLockfile.ts#L81-L82

https://github.com/pnpm/pnpm/blob/b78774ea0f92ccba8f0ee690d88fe39ccfcec7dd/pkg-manager/resolve-dependencies/src/updateLockfile.ts#L221-L223

I'll comment that this is happening in `resolve-dependencies`.